command/metadata.go

```diff
@@ -0,0 +1,125 @@
+package command
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/mackerelio/mackerel-agent/config"
+	"github.com/mackerelio/mackerel-agent/metadata"
+)
+
+func metadataGenerators(conf *config.Config) []*metadata.Generator {
+	generators := make([]*metadata.Generator, 0, len(conf.MetadataPlugins))
+
+	workdir := os.Getenv("MACKEREL_PLUGIN_WORKDIR")
+	if workdir == "" {
+		workdir = os.TempDir()
```

Comment:
I'm not sure sub-directory is better. Can it cause another problem due to invalid permission for the directory? Well, check plugin uses `os.TempDir` for reference. https://github.com/mackerelio/go-check-plugins/blob/6481e313ae25776b1dc052490dbc4bddcb50cf73/check-log/check-log.go#L129-L132